pkg/worker/lifecycle.go

```diff
@@ -364,6 +364,7 @@ func (s *Worker) getContainerEnvironment(request *types.ContainerRequest, option
 		fmt.Sprintf("CONTAINER_ID=%s", request.ContainerId),
 		fmt.Sprintf("BETA9_GATEWAY_HOST=%s", os.Getenv("BETA9_GATEWAY_HOST")),
 		fmt.Sprintf("BETA9_GATEWAY_PORT=%s", os.Getenv("BETA9_GATEWAY_PORT")),
+		fmt.Sprintf("BETA9_GATEWAY_TLS=%s", os.Getenv("BETA9_GATEWAY_TLS")),
```

Comment:
I think we can just detect if a host has TLS enabled programatically, something like this: 

```
def is_tls_enabled(url: str) -> bool:
    """Check if a given URL supports TLS.

    Args:
        url (str): The URL to check.

    Returns:
        bool: True if TLS is enabled, False otherwise.
    """
    try:
        # Extract hostname and default to port 443 for HTTPS
        hostname = url.split("//")[-1].split("/")[0]
        port = 443

        # Create SSL context and test the connection
        context = ssl.create_default_context()
        with socket.create_connection((hostname, port), timeout=5) as sock:
            with context.wrap_socket(sock, server_hostname=hostname):
                return True
    except (ssl.SSLError, socket.error):
        return False
```

Reply:
this makes sense. the code here works